### ISSUE TITLE
Properties are not always ignored in the `id-length` rule

### ISSUE THREAD

**Issue by mathieumg** [eslint/eslint#4347]
1. **The version of ESLint you are using**
  
  1.8.0

2. **What you did (the source code and ESLint configuration)**

  ```js
  myObj.env = true;
  ```

  with the following configuration:

  ```js
    "id-length": [2, {
        "min": 4,
        "properties": "never",
    }],
  ```
  
3. **The actual ESLint output complete with numbers**
  
  `error  Identifier name 'env' is too short. (< 4)  id-length`

4. **What you expected to happen instead**

  No error. Since `"properties"` is set to `"never"`, this error should not occur.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Reproducible in online demo. Looks like a bug.

**Comment by mysticatea** [comment]
I'm working on this.
